### PR TITLE
[PDI-14077] Hadoop File Output fails to load XML

### PR DESCRIPTION
**Pull request by stanislau-strelchanka**

### CODE REVIEW

**Comment by stanislau-strelchanka** [comment]
@m-a-hall, @mattyb149, @mchen-len-son please review

File: src/org/pentaho/di/trans/steps/hadoopfileoutput/HadoopFileOutputMeta.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ protected String loadSource( Node stepnode, IMetaStore metastore ) {
       if ( url != null && !url.startsWith( HadoopSpoonPlugin.MAPRFS_SCHEME ) ) {
         url = HadoopSpoonPlugin.MAPRFS_SCHEME + "://" + url;
       }
-    } else if ( !url.startsWith( HadoopSpoonPlugin.MAPRFS_SCHEME ) ) {
+    } else if ( url != null && !url.startsWith( HadoopSpoonPlugin.MAPRFS_SCHEME ) ) {
```
Comment (mchen-len-son) [inline src/org/pentaho/di/trans/steps/hadoopfileoutput/HadoopFileOutputMeta.java]: looks like null check can be extracted

File: src/org/pentaho/di/trans/steps/hadoopfileoutput/HadoopFileOutputMeta.java [hunk 2]
```diff
@@ -119,7 +119,7 @@ protected String loadSourceRep( Repository rep, ObjectId id_step ) throws Kettle
...
```
Comment (mchen-len-son) [inline src/org/pentaho/di/trans/steps/hadoopfileoutput/HadoopFileOutputMeta.java]: same here

Comment (mchen-len-son) [inline src/org/pentaho/di/trans/steps/hadoopfileoutput/HadoopFileOutputMeta.java]: duplicate code: can it be extracted and reused?

**Comment by stanislau-strelchanka** [comment]
@m-a-hall, @mattyb149, @mchen-len-son updated, please review

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. ant -Dpentaho.coding.standards.version=1.0.4 -Djunit.jvmargs=-XX:PermSize=256m -f build.xml clean-all resolve jacoco publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  src/org/pentaho/di/trans/steps/hadoopfileoutput/HadoopFileOutputMeta.java
  test-src/org/pentaho/trans/steps/hadoopfileoutput/HadoopFileOutputTest.java

Unit test coverage change:
============
org.pentaho.di.core.namedcluster.NamedClusterManager:
 * Branch Change :  + 10.00%
 * Complexity Change :  + 7.84%
 * Instruction Change :  + 9.34%
 * Line Change :  + 13.27%
 * Method Change :  + 18.75%

org.pentaho.di.trans.steps.hadoopfileoutput.HadoopFileOutputMeta:
 * Branch Change :  + 37.50%
 * Complexity Change :  + 23.21%
 * Instruction Change :  + 41.79%
 * Line Change :  + 39.19%
 * Method Change :  + 24.55%

**Comment by pamval** [comment]
@mchen-len-son  please review

File: src/org/pentaho/di/trans/steps/hadoopfileoutput/HadoopFileOutputMeta.java [hunk 3]
```diff
@@ -112,17 +101,21 @@ protected String loadSourceRep( Repository rep, ObjectId id_step ) throws Kettle
...
-        url = HadoopSpoonPlugin.MAPRFS_SCHEME + "://" + url;
-      }
-    } else if ( !url.startsWith( HadoopSpoonPlugin.MAPRFS_SCHEME ) ) {
-      return namedClusterManager.processURLsubstitution( sourceConfigurationName, url, HadoopSpoonPlugin.HDFS_SCHEME,
-          rep.getMetaStore(), new Variables() );
+    return getUrl( rep.getMetaStore(), url, c );
+  }
+
+  private String getUrl( IMetaStore metastore, String url, NamedCluster c ) {
+      if ( c != null && c.isMapr() ) {
```
Comment (mchen-len-son) [inline src/org/pentaho/di/trans/steps/hadoopfileoutput/HadoopFileOutputMeta.java]: @stanislau-strelchanka please, fix the indentation

**Comment by mchen-len-son** [comment]
the rest looks good to me

@stanislau-strelchanka why we need this fix in 5.4 only?

**Comment by stanislau-strelchanka** [comment]
@mchen-len-son no error in 7.0 and 6.1. 5.4 differs from 7 and 6 in some extra checks so fixed  
it in a separate commit.

**Comment by stanislau-strelchanka** [comment]
@m-a-hall, @mattyb149, @mchen-len-son please review

**Comment by stanislau-strelchanka** [comment]
@m-a-hall, @mattyb149, @mchen-len-son please review

**Pull request merged**